AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -40,39 +41,110 @@ def __init__(self, sc_editor):
+        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
+
+        returns: Item selection reference for the newly created variable
+        """
+        self.__validate_new_variable(new_variable_type)
+        self.__click_on_add_variable_button()
+        self.__select_variable_type_from_create_menu(new_variable_type)
+        graph_vars_page = self.__set_new_variable_name(variable_name)
+        selection = self.__get_variable_selection_index(graph_vars_page)
+
+        return selection
+
     def __validate_new_variable(self, new_variable_type: str) -> None:
         """
-        function for checking the provided variable type for validity
+        function for checking the provided variable type validity
 
         returns: None
         """
         assert type(new_variable_type) is str, "Wrong parameter type provided. new_variable_type was not str"
         assert new_variable_type in VARIABLE_TYPES_DICT, \
             "Wrong type provided. new_variable_type is not a valid type"
 
-    def create_new_variable(self, new_variable_type: str) -> None:
+    def __click_on_add_variable_button(self) -> None:
         """
-        Function for adding a new variable to the variable manager's list
+        helper function for adding a new variable. Clicks on the 'Create Variable' button and waits for the UI to render
 
-        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
-
-        returns: None
+        returns None
         """
-        self.__validate_new_variable(new_variable_type)
-
         add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, ADD_BUTTON_QT)
         add_new_variable_button.click()  # Click on Create Variable button
 
         helper.wait_for_condition((
             lambda: self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT) is not None),
             WAIT_TIME_SEC_3)
 
+    def __select_variable_type_from_create_menu(self, new_variable_type: str) -> None:
+        """
+        helper function for adding a new variable. selects the desired variable type from the menu that renders after
+        the create variable button is clicked. waits for the UI to respond after making a selection
+
+        params: The type of variable to create as a string. ie "Boolean"
+
+        returns none
+        """
+
         # Select variable type
         table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
         variable_entry = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
 
-        # Click on it to create variable
+        # Click on it to create variable and wait 1 second for the UI to respond
         pyside_utils.item_view_index_mouse_click(table_view, variable_entry)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
```

Comment:
The steps here don't alight with the method name.

Additionally, that wait is alarming since we're not actually testing for anything here. Why aren't we performing an actual validation? What happens if the UI is running slightly slow?

If there's a legitimate reason for doing this, leave a comment with the GHI to the right of the statement to track the way to resolve this bad pattern.

Reply:
I believe I have strengthened this with helper wait for condition checks against the number of entries in the table. I put a 3 second wait in similar to most other scripts where we use that function.